### ISSUE TITLE
Unable to fetch logs > 14d

### ISSUE THREAD

**Issue by gmavadiy** [grafana/loki#6157]
**Describe the bug**
Unable to query for logs that are older than 14d.  

**To Reproduce**
1. Loki v2.3.0
2. Query for logs > 14d

UI returns:
`invalid query, through < from (2022-04-27 06:59:00 +0000 UTC < 2022-04-28 17:37:00.857446481 +0000 UTC m=-954420.075368366)`

S3 lifecycle i set to delete objects only after 1y

**Expected behavior**
Log lines should be returned

**Environment:**
 - Infrastructure: Kubernetes
 - Deployment tool: helm

**Screenshots, Promtail config, or terminal output**

```
    table_manager:
      retention_deletes_enabled: false
      # to maintain table entries for 365d
      retention_period: 8760h
```

**Comment by Mario-Hofstaetter** [comment]
Go to your running loki servers config endpoint (e.g. <http://your-loki-server:3100/config> ) and copy the complete config and paste it here in the issue (best edit your 1st post)

**Comment by gmavadiy** [comment]
thanks @Mario-Hofstaetter . Looks like couple of things happened here:
1. even though I have retention_deletes_enabled: false (master switch), retention_period cleared the table entries. I fixed that to match retention_period with data retention
2. I matched reject_old_samples_max_age and max_look_back_period to data retention.

After these 2 steps I do see logs

**Comment by gmavadiy** [comment]
Closing the issue for now. Updated comment with the fix. Will open a separate issue for `retention_deletes_enabled` behavior
